[Bug]: AccessDeniedException for lambda/GetFunctionCodeSigningConfig in eu-central-2

### Terraform Core Version

1.3.5

### AWS Provider Version

4.40.0

### Affected Resource(s)

aws_lambda

### Expected Behavior

It should be created and terraform should return success

### Actual Behavior

Lambda is being created and error is being returned. It's not possible to apply changes or destroy the lambda

### Relevant Error/Panic Output Snippet

```shell
[DEBUG] [aws-sdk-go] DEBUG: Response lambda/GetFunctionCodeSigningConfig Details:
---[ RESPONSE ]--------------------------------------
HTTP/2.0 403 Forbidden
X-Amzn-Errortype: AccessDeniedException
[DEBUG] [aws-sdk-go] DEBUG: Validate Response lambda/GetFunctionCodeSigningConfig failed, attempt 0/25, error AccessDeniedException:
status code: 403, request id: xxxx
[DEBUG] Unlocking "aws_lambda_function"
[DEBUG] Unlocked "aws_lambda_function"

Response contains error diagnostic: diagnostic_detail= tf_proto_version=5.3 diagnostic_summary="error getting Lambda Function (test_lambda) code signing config AccessDeniedException:
    status code: 403, request id: xxx" tf_provider_addr=registry.terraform.io/hashicorp/aws tf_req_id=9c3edc87-a488-26bc-cee0-f8b0289a6c22 tf_resource_type=aws_lambda_function @caller=github.com/hashicorp/terraform-plugin-go@v0.14.1/tfprotov5/internal/diag/diagnostics.go:55 @module=sdk.proto diagnostic_severity=ERROR tf_rpc=ApplyResourceChange timestamp=2022-11-23T18:23:31.673+0100
2022-11-23T18:23:31.682+0100 [ERROR] vertex "aws_lambda_function.hello_world" error: error getting Lambda Function (test_lambda) code signing config AccessDeniedException:
    status code: 403, request id: xxx
```


### Terraform Configuration Files
```
terraform {
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "4.40.0"
    }
  }
}

provider "aws" {
  region = "eu-central-2"
}

provider "archive" {}
data "archive_file" "lambda_zip" {
  type        = "zip"
  source_dir  = "./lambdas/test_lambda/"
  output_path = "lambda_function.zip"
}

data "aws_iam_policy_document" "policy" {
  statement {
    sid    = ""
    effect = "Allow"
    principals {
      identifiers = ["lambda.amazonaws.com"]
      type        = "Service"
    }
    actions = ["sts:AssumeRole"]
  }
}

resource "aws_iam_role" "iam_for_lambda" {
  name               = "iam_for_lambda"
  assume_role_policy = data.aws_iam_policy_document.policy.json
}

resource "aws_lambda_function" "hello_world" {
  filename         = data.archive_file.lambda_zip.output_path
  function_name    = "test_lambda"
  role             = aws_iam_role.iam_for_lambda.arn
  handler          = "index.lambda_handler"
  source_code_hash = data.archive_file.lambda_zip.output_base64sha256
  runtime          = "nodejs16.x"
}
```

### Steps to Reproduce

1. Try to created lambda with zipped code in eu-central-2 (Zürich). It works correctly in other regions.


### Debug Output

_No response_

### Panic Output

_No response_

### Important Factoids

The user has full admin permissions, I tried adding:
```
"lambda:UpdateCodeSigningConfig",
"lambda:GetFunctionEventInvokeConfig",
"lambda:GetCodeSigningConfig",
"lambda:CreateCodeSigningConfig",
"lambda:GetFunctionCodeSigningConfig"
```

explicitly, it makes no difference.

### References

I expect that it is the same issue as was described here: https://github.com/hashicorp/terraform-provider-aws/issues/16755 but for a new region.

### Would you like to implement a fix?

_No response_

--- a/.changelog/28008.txt
+++ b/.changelog/28008.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+resource/aws_lambda_function: Don't fail resource Create if AWS Signer service is not available in the configured Region
+```

--- a/internal/service/lambda/function.go
+++ b/internal/service/lambda/function.go
@@ -903,7 +903,7 @@ func resourceFunctionRead(d *schema.ResourceData, meta interface{}) error {
 	// so we cannot just ignore the error as would typically.
 	// We are hardcoding the region here, because go aws sdk endpoints
 	// package does not support Signer service
-	if region := meta.(*conns.AWSClient).Region; region == endpoints.ApNortheast3RegionID || region == endpoints.MeCentral1RegionID || region == endpoints.ApSoutheast3RegionID {
+	if region := meta.(*conns.AWSClient).Region; region == endpoints.ApNortheast3RegionID || region == endpoints.ApSoutheast3RegionID || region == endpoints.EuCentral2RegionID || region == endpoints.MeCentral1RegionID {
 		return nil
 	}
 

--- a/internal/service/lambda/function.go
+++ b/internal/service/lambda/function.go
@@ -898,12 +898,9 @@ func resourceFunctionRead(d *schema.ResourceData, meta interface{}) error {
 		return nil
 	}
 
-	// Currently, this functionality is not enabled in ap-northeast-3 (Osaka), me-central-1 (UAE) and ap-southeast-3 (Jakarta) region
-	// and returns ambiguous error codes (e.g. AccessDeniedException)
-	// so we cannot just ignore the error as would typically.
-	// We are hardcoding the region here, because go aws sdk endpoints
-	// package does not support Signer service
-	if region := meta.(*conns.AWSClient).Region; region == endpoints.ApNortheast3RegionID || region == endpoints.ApSoutheast3RegionID || region == endpoints.EuCentral2RegionID || region == endpoints.MeCentral1RegionID {
+	// Currently this functionality is not enabled in all Regions and returns ambiguous error codes
+	// (e.g. AccessDeniedException), so we cannot just ignore the error as we would typically.
+	if !SignerServiceIsAvailable(meta.(*conns.AWSClient).Region) {
 		return nil
 	}
 
@@ -1446,6 +1443,36 @@ func waitForFunctionUpdate(conn *lambda.Lambda, functionName string, timeout tim
 	_, err := stateConf.WaitForState()
 
 	return err
+}
+
+// SignerServiceIsAvailable returns whether the AWS Signer service is available in the specified AWS Region.
+// The AWS SDK endpoints package does not support Signer.
+// See https://docs.aws.amazon.com/general/latest/gr/signer.html#signer_lambda_region.
+func SignerServiceIsAvailable(region string) bool {
+	availableRegions := map[string]struct{}{
+		endpoints.UsEast2RegionID:      {},
+		endpoints.UsEast1RegionID:      {},
+		endpoints.UsWest1RegionID:      {},
+		endpoints.UsWest2RegionID:      {},
+		endpoints.AfSouth1RegionID:     {},
+		endpoints.ApSouth1RegionID:     {},
+		endpoints.ApNortheast2RegionID: {},
+		endpoints.ApSoutheast1RegionID: {},
+		endpoints.ApSoutheast2RegionID: {},
+		endpoints.ApNortheast1RegionID: {},
+		endpoints.CaCentral1RegionID:   {},
+		endpoints.EuCentral1RegionID:   {},
+		endpoints.EuWest1RegionID:      {},
+		endpoints.EuWest2RegionID:      {},
+		endpoints.EuSouth1RegionID:     {},
+		endpoints.EuWest3RegionID:      {},
+		endpoints.EuNorth1RegionID:     {},
+		endpoints.MeSouth1RegionID:     {},
+		endpoints.SaEast1RegionID:      {},
+	}
+	_, ok := availableRegions[region]
+
+	return ok
 }
 
 func flattenEnvironment(apiObject *lambda.EnvironmentResponse) []interface{} {

--- a/internal/service/lambda/function_test.go
+++ b/internal/service/lambda/function_test.go
@@ -206,16 +206,8 @@ func TestAccLambdaFunction_unpublishedCodeUpdate(t *testing.T) {
 }
 
 func TestAccLambdaFunction_codeSigning(t *testing.T) {
-	if got, want := acctest.Partition(), endpoints.AwsUsGovPartitionID; got == want {
-		t.Skipf("Lambda code signing config is not supported in %s partition", got)
-	}
-
-	// We are hardcoding the region here, because go aws sdk endpoints
-	// package does not support Signer service
-	for _, want := range []string{endpoints.ApNortheast3RegionID, endpoints.ApSoutheast3RegionID, endpoints.EuCentral2RegionID, endpoints.MeCentral1RegionID} {
-		if got := acctest.Region(); got == want {
-			t.Skipf("Lambda code signing config is not supported in %s region", got)
-		}
+	if curr := acctest.Region(); !tflambda.SignerServiceIsAvailable(curr) {
+		t.Skipf("Lambda code signing config is not supported in %s region", curr)
 	}
 
 	var conf lambda.GetFunctionOutput

--- a/internal/service/lambda/function_test.go
+++ b/internal/service/lambda/function_test.go
@@ -212,7 +212,7 @@ func TestAccLambdaFunction_codeSigning(t *testing.T) {
 
 	// We are hardcoding the region here, because go aws sdk endpoints
 	// package does not support Signer service
-	for _, want := range []string{endpoints.ApNortheast3RegionID, endpoints.ApSoutheast3RegionID} {
+	for _, want := range []string{endpoints.ApNortheast3RegionID, endpoints.ApSoutheast3RegionID, endpoints.EuCentral2RegionID, endpoints.MeCentral1RegionID} {
 		if got := acctest.Region(); got == want {
 			t.Skipf("Lambda code signing config is not supported in %s region", got)
 		}

